### PR TITLE
Feature/friend list

### PR DESCRIPTION
**Pull request by Tetardfurieux**

### CODE REVIEW

File: app/src/main/java/ch/epfl/sdp/healthplay/friendlist/AddFriendFragment.java [hunk 1]
```diff
@@ -165,17 +148,31 @@ public void onItemClick(AdapterView<?> parent, View view, int position, long id)
...
-    private void buildListView(View view, ListView listView, List<Friend> friendList) {
-        ArrayList<Friend> arrayOfUsers = new ArrayList<Friend>(friendList);
+    /**
+     * Build the ListView with the list of Friend
+     * @param view
+     * @param listView
+     * @param friendList
+     */
+    public void buildListView(View view, ListView listView, List<Friend> friendList) {
+        ArrayList<Friend> arrayOfUsers = new ArrayList<>(friendList);
```
Comment (ozan-guven) [inline app/src/main/java/ch/epfl/sdp/healthplay/friendlist/AddFriendFragment.java:161]: You could try to use a more general type for this variable as we have seen in the first semester. You could try to change this to List<Friend> arrayOfUsers = new ArrayList<>(friendList) so it's easier to maintain.

File: app/src/main/java/ch/epfl/sdp/healthplay/friendlist/AddFriendFragment.java [hunk 2]
```diff
@@ -165,17 +148,31 @@ public void onItemClick(AdapterView<?> parent, View view, int position, long id)
...
+
+        // Sort Friend by their name
+        Collections.sort(arrayOfUsers, new Comparator<Friend>() {
+            @Override
+            public int compare(Friend m1, Friend m2) {
+                return (int) m1.getUserName().charAt(0) - (int) m2.getUserName().charAt(0);
```
Comment (ozan-guven) [inline app/src/main/java/ch/epfl/sdp/healthplay/friendlist/AddFriendFragment.java:167]: I am not sure about your comparator. It only sorts friends with respect to the fist letter of their name. Is this what you want to do ? 
You could try using the already existing string comparators to compare the name of the friends maybe.

**Review by ozan-guven — CHANGES_REQUESTED** [review]
Very clean code in general. You can look at my comments.

File: app/src/main/java/ch/epfl/sdp/healthplay/navigation/FragmentNavigation.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import ch.epfl.sdp.healthplay.R;
+
+public class FragmentNavigation {
+
+    /**
+     * Switch to the given fragment
+     * @param fragment
+     */
+    public static void switchToFragment(FragmentManager fragmentManager, Fragment fragment){
+        if(fragmentManager != null && fragment != null) {
```
Comment (Zephyr75) [inline app/src/main/java/ch/epfl/sdp/healthplay/navigation/FragmentNavigation.java:15]: You didn't handle the case where either one is null

File: app/src/main/java/ch/epfl/sdp/healthplay/friendlist/AddFriendFragment.java (hunk 2 above)
Comment (Zephyr75) [reply app/src/main/java/ch/epfl/sdp/healthplay/friendlist/AddFriendFragment.java:167]: public class CustomComparator implements Comparator<Friend> {
    @Override
    public int compare(Friend f1, Friend f2) {
        return f1.getUserName().compareTo(f2.getUserName());
    }
}

You could try using something like this.

**Review by Zephyr75 — CHANGES_REQUESTED** [review]
Really nice code and tests, really good comments as well, I suggested a couple of changes.

**Review by Zephyr75 — APPROVED** [review]

**Pull request merged**